### ISSUE TITLE
Generic derived class generates compiler error

### ISSUE THREAD

**Issue by ejsolberg** [immutables/immutables#342]
Using version `2.2`
```
import org.immutables.value.Value;

public interface GenericSubType{

    interface A {
        int a();
    }

    @Value.Immutable
    interface B<T> extends A{
        T b();
    }
}
```

produces a generated class that cannot compile:
```
ImmutableB.java:184: error: illegal generic type for instanceof
      if (object instanceof GenericSubType.B<T>) {
```
in the `private void from(Object object)` method

**Comment by elucash** [comment]
Need to be fixed, thank you for reporting this!

**Comment by elucash** [comment]
Released in 2.2.1 Thank you!
